genie-docs/src/docs/asciidoc/_properties.adoc

```diff
@@ -483,6 +483,16 @@ of 2.0
 |300000
 |no
 
+|genie.scripts.agent-launcher-selector.properties-pattern
+|The regular expression that filters which properties get passed to the script.
+|^agent-launcher-selector\..*
```

Comment:
While I understand the desire for flexibility here I worry that we're just asking for trouble or adding complexity to admins by introducing regex. I feel like when I initially saw this pull request it seemed easier to do something like `genie.scripts.agent-launcher-selector.extra-properties` or something and just say anything under that root will be passed to the script. This is similar to what [spring does with hibernate properties](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html#spring.jpa.properties) for example under `spring.jpa.properties.*`

Reply:
The downside of forcing a prefix like that is that this mechanism cannot be used to 'forward' generic properties.

For example, i cannot pass a loglevel change property to a component used by the script, because the loglevel property doesn't match `genie.scripts.agent-launcher-selector....`.

This may not be a big deal for the selector scripts, but this same mechanism is used by the agent configuration service.
And for the agent it seems more important to be able to forward arbitrary properties we dynamically set.

The default configuration for this component seems to hide the complexity (which is there only if needed).
In the most common case, just set `agent-launcher-selector.foo` and it's gonna make its way to the scripts context.
Super easy for canaries, toggles, etc.

I'll add some doc to to clarify usage and tuning.

Reply:
yeah but do what hibernate does and strip the prefix off and pass the properties in as is... e.g. `genie.blah.script.logging.level.debug` with a prefix of `genie.blah.script` would pass in `logging.level.debug` to the final script

Reply:
Isn't that more complexity for whoever is using this?
Rather than setting a property you have to come up with the right prefix, then set it then retrieve a property with a different name in the runtime..

Reply:
Switched to fixed property prefix that gets stripped before forwarding to agents and scripts.